### ISSUE TITLE
Python Emitter should auto-install or detect the missing dependencies for un-branded scenario

### ISSUE THREAD

**Issue by lirenhe** [microsoft/typespec#5169]
### Clear and concise description of the problem

In order to let vscode typespec extension report the missing dependencies to run codegen,  C# Emitter should have the ability to detect the missing dependencies in the target machine for un-branded scenario and return the correct message.

Please check for more details:  https://github.com/microsoft/typespec/issues/5166

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [docs](https://typespec.io/docs/).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by lirenhe** [comment]
Let's try to ensure all dependencies could be auto installed by using npm install.

**Comment by lmazuel** [comment]
@msyyc let's close this issue only when we got Pyodide version released, and we can put the Python version here as a comment.

**Comment by msyyc** [comment]
Yeah, after https://github.com/microsoft/typespec/pull/5289 or https://github.com/microsoft/typespec/pull/4784 merged, we could close this issue with new emitter version commented.

**Comment by msyyc** [comment]
https://www.npmjs.com/package/@typespec/http-client-python/v/0.5.0 is released.
